controllers/provider-azure/pkg/webhook/network/mutate.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network
+
+import (
+	calicov1aplha1 "github.com/gardener/gardener-extensions/controllers/networking-calico/pkg/apis/calico/v1alpha1"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+// calicoNetworkConfigFromNetwork returns the network provider inside the network resource
+func calicoNetworkConfigFromNetwork(network *extensionsv1alpha1.Network) (*calicov1aplha1.NetworkConfig, error) {
```

Comment:
Use https://github.com/gardener/gardener-extensions/pull/162/files#diff-a233c080feedbca5a745e17554599c60R42 instead of re-implementing it.